calculation/sha_calc/sha_calc/directivity/bea20/bea20.py

```diff
@@ -0,0 +1,226 @@
+    T: np.ndarray
+        The GC2 coordinates in km. Equivalent to rx.
+        Must be nX1 number where n is the number of locations at which the model provides a prediction.
+    Smax: List
+        List of 2x1 number where first is the lowest max value and the second is the largest max value for s.
+        Computed from GC2 using nominal strike of the fault.
+    D: float
+        The effective rupture travel width, measured from the hypocenter to the shallowest depth of the rupture plane.
+    Tbot: float
+        The length of the bottom of the rupture plane (projected to the surface) in km.
+    Dbot: float
+        The vertical depth of the bottom of the rupture plane from the ground surface, including Ztor, in km.
+    Rake: float
+        The Rake of the fault
+    Dip: float
+        The Dip of the fault at the hypocenter
+    Period: float
+        Used for specifying the period for a pSA IM
+    type: float, optional
+        0 for type base don strike
+        1 for a strike slip
+        2 for a oblique, reverse or normal
+    """
+
+    # If not specified, determine rupture category from Rake angle
+    if type == 0:
+        if (-30 <= Rake <= 30) or (-180 <= Rake <= -150) or (150 <= Rake <= 180):
+            type = 1
+        else:
+            type = 2
+
+    # Convert U to S
+    # Limit S to the Smax values for positive and negative numbers
+    Smax1, Smax2 = Smax
+    U, S = np.asarray(U).copy(), np.asarray(U).copy()
+    S[U < Smax1] = Smax1
+    S[U > Smax2] = Smax2
+
+    # Convert U to Ry
+    # Reduce each positive and negative side by absolute value of Smax and remove negative numbers inbetween
+    upos, uneg = U >= 0, U < 0
+    Ry = np.zeros(U.size)
+    Ry[upos] = U[upos] - Smax2
+    Ry[uneg] = abs(U[uneg]) - abs(Smax1)
+    Ry[Ry < 0] = 0
+
+    # Calculate S2
+    if type == 2 and Rake < 0:
+        S = -S
+    Srake = S * math.cos(math.radians(Rake))
+    D = max(D, 3)  # Gets the max value for D, 3 is the minimum
+    S2 = np.sqrt(Srake ** 2 + D ** 2)
+
+    # Predictor variable fs2
+    if type == 1:
+        fs2 = np.log(S2)
+        fsCap = math.log(465)
+        fs2[fs2 > fsCap] = fsCap
+    else:
+        fs2 = np.zeros(U.size)
+        spos, sneg = Srake >= 0, Srake < 0
+        fs2[spos] = np.log(S2[spos])
+        fs2[sneg] = np.log(D)
+        fsCap = math.log(188)
+        fs2[fs2 > fsCap] = fsCap
+
+    # Angular predictor variables
+    T = np.asarray(T).copy()
+    if type == 1:
+        theta = np.nan_to_num(np.arctan(np.divide(T, U)))
+        ftheta = abs(np.cos(2 * theta))
+        fphi = np.ones(U.size)
+    else:
+        tpos, tneg = T > 0, T <= 0
+        phi = np.zeros(U.size)
+
+        phi[tneg] = (
+            np.degrees(np.arctan(np.divide(abs(T[tneg]) + Tbot, Dbot))) + Dip - 90
+        )
+
+        t1 = np.logical_and(tpos, T < Tbot)
+        phi[t1] = 90 - Dip - np.degrees(np.arctan(np.divide(Tbot - T[t1], Dbot)))
+
+        t2 = np.logical_and(tpos, T >= Tbot)
```

Comment:
I personally prefer using `&`/`|` for logical and/or, i.e.
`t2 = tpos & (T >= Tbot)` (just be sure to have the parentheses)

But that's a personal preference, so up to you